crates/subspace-farmer/src/bin/subspace-farmer/main.rs

```diff
@@ -91,6 +91,15 @@ struct FarmingArgs {
     /// one specified endpoint. Format: 127.0.0.1:8080
     #[arg(long, alias = "metrics-endpoint")]
     metrics_endpoints: Vec<SocketAddr>,
+    /// Defines how many sectors farmer will download concurrently, allows to limit memory usage of
+    /// the plotting process, must always be larger than `encoding_semaphore` or else plotting may
+    /// deadlock
+    #[arg(long, default_value_t = 2)]
+    sector_downloading_concurrency: usize,
```

Comment:
What do you think about introducing either `NonZero` type here or a validation function for these parameters? Setting those parameters to zero leads to guaranteed stalling of the plotting.

Also, what will happen if we set a very large number here? Should we introduce an upper bound as well here?

Reply:
Non-zero makes sense. Very large numver is still bound by the number of farms and will simply result in slower plotting with higher memory usage.